packages/calendar/react/src/components/action-pane/action-pane.tsx

```diff
@@ -0,0 +1,314 @@
+    RouterInputs["eventGroup"]["create"]["events"][number]["attendees"]
+  >([]);
+  const [showActionTypesPopup, setShowActionTypesPopup] = useState(false);
+  const [isCreating, setIsCreating] = useState(false);
+  const [showCreatingSpinner, setShowCreatingSpinner] = useState(false);
+
+  const [buttonType, setButtonType] = useState<ActionType>("Copy and Close");
+
+  const { data: preference } = trpc.preference.byCurrentUser.useQuery();
+  const { mutate: createEventGroup, isLoading: isCreatingEventGroup } =
+    trpc.eventGroup.create.useMutation({
+      onSuccess() {
+        setUnknownAttendees([]);
+        utils.event.all.invalidate();
+
+        copyToClipboard(selectedSlots, preference);
+        toast("Saved and copied to clipboard!", {
+          position: "bottom-center",
+          hideProgressBar: true,
+          autoClose: 1000,
+          type: "success",
+        });
+        setIsCreating(false);
+        clearSlots();
+        resetTitle();
+      },
+      onError() {
+        toast("Failed to create events", {
+          position: "bottom-center",
+          hideProgressBar: true,
+          autoClose: 2000,
+          type: "error",
+        });
+        setIsCreating(false);
+      },
+    });
+
+  const isDisabled =
+    isCreating || isCreatingEventGroup || selectedSlots.length === 0;
+
+  const handleClose = () => {
+    onClose?.();
+  };
+
+  const handleSave = async () => {
+    setIsCreating(true);
+    setShowCreatingSpinner(true);
+
+    if (buttonType === "Copy and Close") {
+      setTimeout(() => {
+        copyToClipboard(selectedSlots, preference);
+        toast("Saved and copied to clipboard!", {
+          position: "bottom-center",
+          hideProgressBar: true,
+          type: "success",
+          delay: 0,
+        });
+        setShowCreatingSpinner(false);
+
+        setTimeout(() => {
+          setIsCreating(false);
+          resetTitle();
+          clearSlots();
+        }, 1000);
+        onPrimaryActionClick?.(buttonType);
+      }, 500);
+      return;
+    }
+
+    // Save events
+
+    // Create events in real calendars
+    createEventGroup({
+      createBlocker: true,
+      title: selectedSlots[0]?.title || "",
+      events: selectedSlots.map((slot) => {
+        if (!slot.start || !slot.end) {
+          throw new Error("Please select a correct slot");
+        }
+        return {
+          title: slot.title || "",
+          startDate: convertToDate(slot.start),
+          endDate: convertToDate(slot.end),
+          attendees: unknownAttendees.concat(
+            directoryUsersWithEvents.map((u) => ({
+              id: u.id,
+              name: u.name,
+              surname: u.surname,
+              email: u.email,
+              provider: u.provider,
+              responseStatus: EventResponseStatus.NEEDS_ACTION,
+            })),
+          ),
+        };
+      }),
+    });
+  };
+
+  const actionTypesPopup = (
+    <OutsideClickHandler
```

Comment:
would be nice to find a replacement for this component too. kinda old and deprecated. doesn't radix have something here?